src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        * `MeshSurfaceRegion`
+        * `PolygonalFootprintRegion`
+        """
+        # Check if region is fixed, and if not returns a default implementation
+        if needsSampling(self):
+            return super().intersects(other)
+
+        elif isinstance(other, MeshSurfaceRegion):
+            # Uses Trimesh's collision manager to check for intersection of the
+            # surfaces. Use SurfaceCollisionTrimesh objects to ensure collisions
+            # actually imply a surface collision.
+            collision_manager = trimesh.collision.CollisionManager()
+
+            collision_manager.add_object("SelfRegion", SurfaceCollisionTrimesh(faces=self.mesh.faces, vertices=self.mesh.vertices))
+            collision_manager.add_object("OtherRegion", SurfaceCollisionTrimesh(faces=other.mesh.faces, vertices=other.mesh.vertices))
+
+            surface_collision = collision_manager.in_collision_internal()
+
+            return surface_collision
+
+        if isinstance(other, PolygonalFootprintRegion):
+            # Determine the mesh's vertical bounds (adding a little extra to avoid mesh errors) and
+            # the mesh's vertical center.
+            vertical_bounds = (self.mesh.bounds[0][2], self.mesh.bounds[1][2])
+            mesh_height = vertical_bounds[1] - vertical_bounds[0] + 1
+            centerZ = (vertical_bounds[1] + vertical_bounds[0])/2
+
+            # Compute the bounded footprint and recursively compute the intersection
+            bounded_footprint = other.approxBoundFootprint(centerZ, mesh_height)
+
+            return self.intersects(bounded_footprint)
+
+        elif not triedReversed:
+            return other.intersects(self)
+
+        raise NotImplementedError("Cannot check intersection of MeshRegion with " +
+            type(other) + ".")
+
+    def containsPoint(self, point):
+        """Check if this region's surface contains a point."""
+        # First compute the minimum distance to the point.
+        min_distance = self.distanceTo(point)
+
+        # If the minimum distance is within tolerance of 0, the mesh contains the point.
+        return min_distance < self.tolerance
+
+    def containsObject(self, obj):
+        # A surface cannot contain an object, which must have a volume.
+        return False
+
+    def containsRegionInner(self, reg, tolerance):
+        if tolerance != 0:
+            warnings.warn("Nonzero tolerances are ignored for containsRegionInner on MeshSurfaceRegion")
+
+        if isinstance(reg, MeshSurfaceRegion):
+            diff_region = reg.difference(self)
+
+            return isinstance(diff_region, EmptyRegion)
+
+        raise NotImplementedError
+
+    def uniformPointInner(self):
+        return Vector(*trimesh.sample.sample_surface(self.mesh, 1)[0][0])
+
+    def distanceTo(self, point):
+        """Get the minimum distance from this object to the specified point."""
+        point = toVector(point, "Could not convert 'point' to vector.")
```

Comment:
Did you mean this to be an f-string? The user won't know what "point" is.

Reply:
(Looks like this was marked resolved by mistake.)

Reply:
Whoops, I only swapped MeshVolumeRegion, not MeshSurfaceRegion. Fixed now.